### ISSUE TITLE
[Bug] Broken types - @storybook/types issue

### ISSUE THREAD

**Issue by rChaoz** [storybookjs/addon-svelte-csf#194]
### Describe the bug

At some points between 8.0.10 and 8.2.6, something changed with how types are handled internally in the storybook. More specifically, I've seen that generally types are now imported from `storybook/internal/types` rather than `@storybook/types`. The issue here is that it seems like the package `@storybook/types` is not a dependency of other storybook packages anymore, and addon-svelte-csf also doesn't depend on it (only through devDependencies, which are not propagated), so the types package not installed anymore, making any code that tries to import it fails.

More specifically I'm talking about this one line of code:
https://github.com/storybookjs/addon-svelte-csf/blob/35a9b514ea67b6a6a29aac31eeaaca98579738d2/src/index.d.ts#L2

### Steps to reproduce the behavior

1. Any project that uses Storybook 8.2.6 and addon-svelte-csf
2. Make sure to not manually depend on `@storybook/types`
3. Write a story
4. Seemengly random/weird errors:

```
Error: Object literal may only specify known properties, and 'args' does not exist in type 'Partial<StoryProps>'. (ts)
    const controls: Partial<StoryProps> = {
        args: {
            creating: false,
```

The problem is that, inside addon-svelte-csf, `StoryProps` extends `BaseAnnotations` from `@storybook/types`, which declares `args`, `argTypes` and many others. But, as the package is not installed, TypeScript doesn't find the interface it's supposed to extend, making it as if it doesn't extend anything at all.

### Workaround

Manually add `@storybook/types` as a dependency.

**Comment by xeho91** [comment]
Which version of `@storybook/addon-svelte-csf` are you using?

**Comment by rChaoz** [comment]
My bad, forgot to mention. This issue happened after updating everything to the latest version, so that would be 4.1.4.

**Comment by JReinhold** [comment]
@ndelangen do you agree here that the solution is just to move `@storybook/types` to a regular dependency, as we need to support all of 8.X?

note this addon doesn't use the blessed tsup setup due to.. Svelte-related reasons.

**Comment by olafurw** [comment]
Had this happen on our system after an update. I'm using 4.1.4 for CSF and 8.2.7 of Storybook. Workaround fixed it for us.

**Comment by olafurw** [comment]
> This is likely fixed in the latest patch version of storybook @JReinhold

No, like I say in my comment, I'm using 8.2.7 (which is the latest version as I type this) and I need to install the `@storybook/types` library as a workaround to no longer get the errors.

**Comment by JReinhold** [comment]
> This is likely fixed in the latest patch version of storybook @JReinhold 

No this addon still depends on the types package even though the issue is fixed in core. The types package is still missing.

**Comment by shilman** [comment]
:rocket: Issue was released in [`v4.1.5`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v4.1.5) :rocket:
